src/parser/view_transform.c

```diff
@@ -1670,37 +1716,42 @@ mq_substitute_subquery_in_statement (PARSER_CONTEXT * parser, PT_NODE * statemen
 	  PT_INTERNAL_ERROR (parser, "class spec not found");
 	  goto exit_on_error;
 	}
-      else
-	{
-	  /* check for (non-pushable) spec set */
-	  rewrite_as_derived = (class_spec->info.spec.entity_name != NULL
-				&& class_spec->info.spec.entity_name->node_type == PT_SPEC);
-	}
+
+      /* determine if class_spec is the only spec in the statement */
+      is_only_spec = ((statement_spec->next == NULL && pred == NULL) ? true : false);
 
       /* do not rewrite vclass_query as a derived table if spec belongs to an insert statement. */
       if (tmp_result->node_type == PT_INSERT)
 	{
 	  rewrite_as_derived = false;
 	}
+      else if (query_spec->node_type != PT_SELECT)
+	{
+	  rewrite_as_derived = true;
+	}
+      /* check for (non-pushable) spec set (spec set??) */
+      else if (class_spec->info.spec.entity_name != NULL && class_spec->info.spec.entity_name->node_type == PT_SPEC)
+	{
+	  rewrite_as_derived = true;
+	}
+      /* check for CONNECT BY */
+      else if (tmp_result->info.query.q.select.connect_by)
+	{
+	  rewrite_as_derived = true;
+	}
+      /* determine if spec is outer joined */
+      else if (mq_is_outer_join_spec (parser, class_spec))
+	{
+	  rewrite_as_derived = true;
+	}
+      /* determine if vclass_query is pushable */
+      else if (!mq_is_pushable_subquery (parser, query_spec, is_only_spec))
+	{
+	  rewrite_as_derived = true;
+	}
       else
 	{
-	  /* determine if class_spec is the only spec in the statement */
-	  is_only_spec = ((statement_spec->next == NULL && pred == NULL) ? true : false);
-
-	  /* determine if spec is outer joined */
-	  is_outer_joined = mq_is_outer_join_spec (parser, class_spec);
-
-	  /* determine if vclass_query is pushable */
-	  is_pushable_query = mq_is_pushable_subquery (parser, query_spec, is_only_spec);
-
-	  /* rewrite vclass_query as a derived table if spec is outer joined or if query is not pushable */
-	  rewrite_as_derived = rewrite_as_derived || !is_pushable_query || is_outer_joined;
-
-	  if (PT_IS_QUERY (tmp_result))
-	    {
-	      rewrite_as_derived = rewrite_as_derived || (tmp_result->info.query.all_distinct == PT_DISTINCT)
-		|| (PT_IS_VALUE_QUERY (query_spec));
-	    }
+	  rewrite_as_derived = false;
```

Comment:
In the clause "if .... else if ... else", the last "else" and the first "if ... rewirte_as_derived = false" looks not necessary because the variable rewrite_as_derived is already set to "false" initially.

Reply:
I know what you said. But I think it is good for readability.